Adds resultBackendSecretName warning in Helm production docs (#23307)
mistercrunch/Airflow: docs/helm-chart/production-guide.rst

  data:
    metadataSecretName: mydatabase

>     resultBackendSecretName: mydatabase

Reviewer: ```suggestion
```

Let's leave this executor agnostic.

.. _production-guide:pgbouncer:

.. warning::

>   `resultBackendSecretName` expects a url that starts with `db+postgresql://`, while `metadataSecretName` expects `postgresql://` or `postgres://` and wont work with `db+postgresql://`. To solve this, it might be necessary to create multiple secrets.

Reviewer: ```suggestion
  If you use `CeleryExecutor`, keep in mind that `resultBackendSecretName` expects a url that starts with `db+postgresql://`, while `metadataSecretName` expects `postgresql://` and wont work with `db+postgresql://`. You'll need to create separate secrets with the correct schema.
```

Starting in 2.3.0, only `postgresql` works, so lets just keep that one. Otherwise, I think this sounds a little better, but feel free to adjust it more.

.. _production-guide:pgbouncer:

.. warning::

>   If you use `CeleryExecutor`, keep in mind that `resultBackendSecretName` expects a url that starts with `db+postgresql://`, while `metadataSecretName` expects `postgresql://` and wont work with `db+postgresql://`. You'll need to create separate secrets with the correct schema.

Reviewer: ```suggestion
  If you use `CeleryExecutor`, keep in mind that `resultBackendSecretName` expects a url that starts with `db+postgresql://`, while `metadataSecretName` expects `postgresql://` and wont work with `db+postgresql://`. You'll need to create separate secrets with the correct scheme.
```

Author: Should `CeleryKubernetesExecutor` be added as well ?
I notice the docs usually mention only one of them.

Reviewer: We normally just reference CeleryExecutor explicitly, as CeleryKubernetesExecutor actually runs a CeleryExecutor under the hood. Said another way, it's inferred (at least to me).

>   If you use `CeleryExecutor`, keep in mind that `resultBackendSecretName` expects a url that starts with `db+postgresql://`, while `metadataSecretName` expects `postgresql://` and wont work with `db+postgresql://`. You'll need to create separate secrets with the correct scheme.

Reviewer: ```suggestion
  If you use ``CeleryExecutor``, keep in mind that ``resultBackendSecretName`` expects a url that starts with ``db+postgresql://``, while ``metadataSecretName`` expects ``postgresql://`` and won't work with ``db+postgresql://``. You'll need to create separate secrets with the correct scheme.
```
